### PR TITLE
Trim Ci cache before upload, rollback wasmtime to v0.37

### PR DESCRIPTION
**Pull request by mriise**
Wasmtime v0.38 introduced a regression, causing integration test `native_overflow_somethingsomething` to run for waay to long.

Rolling back the dep brought back some CI issues I saw earlier where the cache would keep growing after each build, so now it is trimmed before upload (as well as printing out sccache and cargo cache size)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -117,3 +117,21 @@ jobs:
...
+      uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505 # v1.0.3
+      with:
+        command: cache
+        args: -e -i
+    - name: Getting sccache size
+      if: ${{ matrix.push }}
+      uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505 # v1.0.3
+      with:
+        command: cache
+        args: sccache
```
Comment (Stebalien) [inline .github/workflows/ci.yml:137]: Do we actually need sscache?

Comment (mriise) [reply .github/workflows/ci.yml:143]: not really? but it's not what is causing the cache to grow, its `/.cargo/registry`

Comment (mriise) [reply .github/workflows/ci.yml:143]: it does improve compile times by a bit and stays around 500MB, so I wouldn't remove it just because

**Comment by mriise** [comment]
success! https://github.com/filecoin-project/ref-fvm/runs/7101326285?check_suite_focus=true

**Review by Stebalien — APPROVED** [review]

**Pull request merged**